src/Microsoft.AspNet.Mvc.Core/Rendering/Html/IHtmlGenerator.cs

```diff
@@ -31,6 +31,8 @@ public interface IHtmlGenerator
 
         TagBuilder GenerateAntiForgery([NotNull] ViewContext viewContext);
 
+        void GeneratePresetCookieAndHeaders([NotNull] ViewContext viewContext);
```

Comment:
Waiting for decision on whether this method needs to be here. So will hold off on adding comments until the decision is made.